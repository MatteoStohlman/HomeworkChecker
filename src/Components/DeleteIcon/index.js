import React from 'react';
import {compose,pure} from 'recompose';
import { withStyles } from '@material-ui/core/styles';
import Icon from '@material-ui/core/Icon';

const styles = {
  root:{
    cursor:'pointer'
  }
}

const DeleteIcon = ({
  //PROPS
    //optional
      onClick,iconName='delete',
  //OTHER
    classes,...props
})=> {

  return (
    <Icon onClick={onClick&&onClick} className={classes.root} {...props}>{iconName}</Icon>
  )
}

export default compose(
  pure,
  withStyles(styles)
)(DeleteIcon)
