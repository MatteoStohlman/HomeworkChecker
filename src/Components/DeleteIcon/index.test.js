import React from 'react';
import { mount , shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import DeleteIcon from './index';

describe('Delete Icon',() => {
  test('basic DeleteIcon renders and snapshot', () => {
    shallow(<DeleteIcon/>)
    const DefaultIcon = renderer.create(
      <DeleteIcon/>
    );
    expect(DefaultIcon.toJSON()).toMatchSnapshot();

    //Test case where valid icon name prop is passed
    const CustomIcon = renderer.create(
      <DeleteIcon iconName='close'/>
    );
    expect(CustomIcon.toJSON()).toMatchSnapshot();

    //Test case where invalid icon name prop is passed
    const BadCustomIcon = renderer.create(
      <DeleteIcon iconName='failing'/>
    );
    expect(BadCustomIcon.toJSON()).toMatchSnapshot();
  });

  test('DeleteIcon handles a non function in onClick',() => {
    const icon = mount(<DeleteIcon onClick='green'/>)
    icon.find('DeleteIcon').simulate('click')
  })

  test('DeleteIcon calls back the onClick prop',()=>{
    const mockCallBack = jest.fn();
    const icon = mount((<DeleteIcon onClick={mockCallBack}/>))
    icon.find('DeleteIcon').simulate('click')

    expect(mockCallBack.mock.calls.length).toEqual(1)
  })
})
