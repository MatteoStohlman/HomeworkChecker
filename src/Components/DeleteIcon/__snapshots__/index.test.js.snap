// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Icon basic DeleteIcon renders and snapshot 1`] = `
<span
  aria-hidden={true}
  className="material-icons MuiIcon-root DeleteIcon-root-1"
  onClick={[Function]}
>
  delete
</span>
`;

exports[`Delete Icon basic DeleteIcon renders and snapshot 2`] = `
<span
  aria-hidden={true}
  className="material-icons MuiIcon-root DeleteIcon-root-1"
  onClick={[Function]}
>
  close
</span>
`;

exports[`Delete Icon basic DeleteIcon renders and snapshot 3`] = `
<span
  aria-hidden={true}
  className="material-icons MuiIcon-root DeleteIcon-root-1"
  onClick={[Function]}
>
  failing
</span>
`;
